test/metabase/test/data/impl.clj

```diff
@@ -112,16 +113,23 @@
         (Database (u/get-id db))
         (catch Throwable e
           (db/delete! Database :id (u/get-id db))
-          (throw e))))
+          (throw (ex-info "Failed to create test database"
+                          {:driver             driver
+                           :database-name      database-name
+                           :connection-details connection-details}
+                          e)))))
```

Comment:
ultimately not needed for this PR but I added some more explicit exception context when loading test data